### DISCUSSION TITLE
Even though the CIDR range is in NO_PROXY, HTTP_PROXY still takes effect

### DISCUSSION THREAD

**Discussion by softyoungha** [encode/httpx#3224] [Potential Issue]
I discovered an issue while using httpx in Airflow. 
Contrary to what is documented in httpx https://github.com/encode/httpx/blob/db9072f998b53ff66d50778bf5edee8e2cc8ede1/httpx/_utils.py#L228-L239 , 
when I inputs a CIDR range in no_proxy, I found that the proxy settings were still being used, even for addresses within that range

you can try this

```python
import os
from httpx._utils import get_environment_proxies

os.environ["http_proxy"] = "http://proxy.example.com"
os.environ["https_proxy"] = "http://proxy.example.com"
os.environ["no_proxy"] = "localhost,10.0.0.0/8"

httpx_proxy_info = get_environment_proxies()
print(httpx_proxy_info)
# {
#       'http://': 'http://proxy.example.com',
#       'https://': 'http://proxy.example.com',
#       'all://localhost': None,
#       'all://10.0.0.0/8': None     <- no_proxy
# }

from httpx._utils import URLPattern
from httpx._urls import URL

url_pattern = URLPattern("all://10.0.0.0/8")
print(url_pattern.matches(URL("http://10.0.0.0")))
# True
print(url_pattern.matches(URL("http://10.240.5.168")))
# False
```

After reviewing the code, it seems that it can only interpret bit ranges like /8, /16, /24 as URL delimiter slashes due to its structure. 
I tried to make custom modifications, but it looks like significant changes would be needed within mounts as well, so I gave up

This issue is quite important to me, so I've also raised it in Airflow. 

> https://github.com/apache/airflow/issues/39794

I'm curious if there are any other methods or plans to fix it.

**Comment by drdavella** [comment]
I just encountered the same behavior, it seems like a good idea to open an issue.
